native/src/main/java/io/ballerina/lib/sap/Listener.java

```diff
@@ -30,16 +33,11 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
-
 public class Listener {
 
-    private static final Logger logger = LoggerFactory.getLogger(Client.class);
-
-    private static final ArrayList<BObject> startedServices = new ArrayList<>();
-    private static final boolean started = false;
-    private static ArrayList<BObject> services = new ArrayList<>();
-    private static Runtime runtime;
+    private static final Logger logger = LoggerFactory.getLogger(Listener.class);
+    private static boolean isServiceAttached = false;
+    private static String jcoServerName;
```

Comment:
Having static fields and updating them using the static methods will be an issue here, because when we have more than one listener, both will try to update this static field which will ends up in unexpected behaviours.

The safest approach is to define this class as final and move `isServiceAttached` and `jcoServerName` fields as native data of the Ballerina listener object or store them in the ballerina listener class

Reply:
Addressed with [986e899](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/47/commits/986e8992704374cc5b22f3813ebe72a30b144793).